tests/31sync/15lazy-members.pl

```diff
@@ -807,3 +807,51 @@ sub check_filler_event {
          Future->done(1);
       });
    };
+
+
+test "Rejecing invite over federation doesn't break incremental /sync",
+   requires => [ remote_user_fixture(),
+      do {
+         my $creator = local_user_fixture();
+         $creator, inviteonly_room_fixture( creator => $creator );
+      }
+   ],
+
+   check => sub {
+      my ( $invitee, $creator, $room_id ) = @_;
+
+      my ( $filter_id );
+
+      matrix_create_filter( $invitee, {
+         room => {
+            state => {
+               lazy_load_members => JSON::true,
```

Comment:
Does this really require lazy loading? I'm not really sure I'm following that bit.

Reply:
During lazy loading we pull out state a second time to get the membership event of the sender of the outlier event, so we stick lazy loading on to hit more of the /sync code paths

Reply:
Thanks! 👍